controllers/openshift_controller.go

```diff
@@ -595,212 +388,162 @@ func (r *KataConfigOpenShiftReconciler) processKataConfigDeleteRequest() (ctrl.R
+				"total machines", parentMcp.Status.MachineCount)
+			if parentMcp.Status.ReadyMachineCount != parentMcp.Status.MachineCount {
 				return ctrl.Result{Requeue: true, RequeueAfter: 15 * time.Second}, nil
 			}
-		}
 
-		for _, nodeName := range r.kataConfig.Status.UnInstallationStatus.InProgress.BinariesUnInstalledNodesList {
-			if contains(r.kataConfig.Status.UnInstallationStatus.Completed.CompletedNodesList, nodeName) {
-				continue
+			mcp := r.newMCPforCR()
+			err = r.Client.Delete(context.TODO(), mcp)
+			if err != nil {
+				// error during removing mcp, don't block the uninstall. Just log the error and move on.
+				r.Log.Info("Error found deleting mcp. If the mcp exists after installation it can be safely deleted manually.",
+					"mcp", mcp.Name, "error", err)
 			}
 
-			r.kataConfig.Status.UnInstallationStatus.Completed.CompletedNodesCount++
-			r.kataConfig.Status.UnInstallationStatus.Completed.CompletedNodesList = append(r.kataConfig.Status.UnInstallationStatus.Completed.CompletedNodesList, nodeName)
-			if r.kataConfig.Status.UnInstallationStatus.InProgress.InProgressNodesCount > 0 {
-				r.kataConfig.Status.UnInstallationStatus.InProgress.InProgressNodesCount--
+			mc, err := r.newMCForCR(machinePool)
+			err = r.Client.Delete(context.TODO(), mc)
+			if err != nil {
+				// error during removing mc, don't block the uninstall. Just log the error and move on.
+				r.Log.Info("Error found deleting machine config. If the machine config exists after installation it can be safely deleted manually.",
+					"mc", mc.Name, "error", err)
 			}
-		}
-
-		err = r.Client.Status().Update(context.TODO(), r.kataConfig)
-		if err != nil {
-			return ctrl.Result{}, err
-		}
-
-		r.Log.Info("Deleting uninstall daemonset")
-		err = r.deleteKataDaemonset(UninstallOperation)
-		if err != nil {
-			return ctrl.Result{}, err
-		}
-
-		r.Log.Info("Uninstallation completed on all nodes. Proceeding with the KataConfig deletion")
-		controllerutil.RemoveFinalizer(r.kataConfig, kataConfigFinalizer)
-		err = r.Client.Update(context.TODO(), r.kataConfig)
-		if err != nil {
-			return ctrl.Result{}, err
+		} else {
+			return ctrl.Result{Requeue: true, RequeueAfter: 15 * time.Second}, nil
 		}
 	}
-	return ctrl.Result{}, nil
-}
-
-func (r *KataConfigOpenShiftReconciler) deleteKataDaemonset(operation DaemonOperation) error {
 
-	ds := r.processDaemonsetForCR(operation)
-	foundDs := &appsv1.DaemonSet{}
-	err := r.Client.Get(context.TODO(), types.NamespacedName{Name: ds.Name, Namespace: ds.Namespace}, foundDs)
-	if err != nil && errors.IsNotFound(err) {
-		// DaemonSet not found, nothing to delete, ignore the request.
-		return nil
-	} else if err != nil {
-		return err
+	r.kataConfig.Status.UnInstallationStatus.InProgress.IsInProgress = corev1.ConditionFalse
+	_, result, err2, done := r.updateStatus(machinePool)
+	r.clearInstallStatus()
```

Comment:
r.clearUnInstallStatus() ?

Reply:
This is intentional to clear the status of the last installation now that the uninstallation is done as it is not relevant any more.